core/shared/src/main/scala/scodec/bits/ByteVector.scala

```diff
@@ -1247,6 +1270,19 @@ object ByteVector {
     override def copyToStream(s: OutputStream, offset: Long, size: Long): Unit = {
       s.write(arr, offset.toInt, toIntSize(size))
     }
+
+    override def copyToBuffer(buffer: ByteBuffer, offset: Long, size: Int): Unit = {
+      val toCopy: Array[Byte] = {
+        if (offset == 0 && size == arr.size) arr
+        else {
+          val tmp = Array.ofDim[Byte](size)
+          copyToArray(tmp, 0, offset, size)
```

Comment:
Do we need the extra copy here?
Would calling `buffer.put(arr, offset, size)` suffice?